language/move-lang/src/expansion/aliases.rs

```diff
@@ -3,64 +3,63 @@
 
 use crate::{
     expansion::ast::ModuleIdent,
-    shared::{remembering_unique_map::RememberingUniqueMap, *},
+    shared::{unique_map::UniqueMap, unique_set::UniqueSet, *},
 };
 use move_ir_types::location::*;
-use std::{collections::BTreeSet, iter::IntoIterator};
+
+type ScopeDepth = usize;
 
 #[derive(Clone, Debug)]
 pub struct AliasSet {
-    pub modules: BTreeSet<Name>,
-    pub members: BTreeSet<Name>,
+    pub modules: UniqueSet<Name>,
+    pub members: UniqueSet<Name>,
+}
+
+#[derive(Clone, Debug)]
+pub struct AliasMapBuilder {
+    modules: UniqueMap<Name, (ModuleIdent, /* is_implicit */ bool)>,
+    members: UniqueMap<Name, ((ModuleIdent, Name), /* is_implicit */ bool)>,
```

Comment:
I've read a lot of Internet armchair opinions recently on using enums vs. booleans, and while I discount most of the stuff I read on the Internet, I do wonder whether an enum would make more sense here.

Most spots in this file where `true` or `false` is passed into these tuples are also accompanied by a comment, `/* is_implicit */`. An alternative would be an `enum AliasKind { Explicit, Implicit }`, which would allow you to pass in `Implicit` instead. Readers of the code would know what that means, without needing to read a code comment, or add their own if they modify this file.

(And, should an additional alias "kind" be added to the language later, it'd be easier to extend the `enum` than it would be to modify the `bool` use sites.)

Reply:
I'd definitely agree with you if the flag was public. 
But as it is internal, I don't think the weight of an enum is worth it. The limited usage internally makes this fine for me
```
if !is_implicit { ... }
```
is a lot more ergonomic to me than 
```
if matches(implicitness, AliasKind::Explicit) { ... }
```
And I really hate this because the match isn't explicit... so I would probably end up doing
```
match implicitness {
  AK::Implicit =>  (),

}
```


This would be different if the usages were always 
```
if is_implicit { /* case 1 * } else { /* case 2 */ }
```
since 
```
match implicitness {
  AK::Implicit => /* case 1 */,
  AK::Explicit => /* case 2 */,
}
```
would be much more clear

tldr: it is an internal flag that is only ever used for a single branch so I feel an enum would make things more confusing in many places

Reply:
So maybe a better summary of I use enums like this:
- Enums are great for flags for public APIs
- Enums are great for flags when you want to do something in each case
- Enums not so great when used for guard-style programming (as I tend to dislike non-exhaustive matches, option is an exception usually since it is always and forever only have two cases)

Reply:
Makes sense to me!